src/libDirectoryService/ShardSizeCaluator.cpp

```diff
@@ -0,0 +1,130 @@
+/**
+* Copyright (c) 2018 Zilliqa 
+* This source code is being disclosed to you solely for the purpose of your participation in 
+* testing Zilliqa. You may view, compile and run the code for that purpose and pursuant to 
+* the protocols and algorithms that are programmed into, and intended by, the code. You may 
+* not do anything else with the code without express permission from Zilliqa Research Pte. Ltd., 
+* including modifying or publishing the code (or any part of it), and developing or forming 
+* another public or private blockchain network. This source code is provided ‘as is’ and no 
+* warranties are given as to title or non-infringement, merchantability or fitness for purpose 
+* and, to the extent permitted by law, all liability for your use of the code is disclaimed. 
+* Some programs in this code are governed by the GNU General Public License v3.0 (available at 
+* https://www.gnu.org/licenses/gpl-3.0.en.html) (‘GPLv3’). The programs that are governed by 
+* GPLv3.0 are those programs that are located in the folders src/depends and tests/depends 
+* and which include a reference to GPLv3 in their program files.
+**/
+
+#include "DirectoryService.h"
+
+uint32_t DirectoryService::CalculateShardSize(const uint32_t numberOfNodes)
+{
+    if (numberOfNodes < 651)
+    {
+        return 600;
+    }
+    else if (numberOfNodes >= 651)
+    {
+        return 651;
+    }
+    else if (numberOfNodes >= 1368)
+    {
+        return 684;
+    }
+    else if (numberOfNodes >= 2133)
```

Comment:
For all cases, it will fall under only one of two if statements. Ex. numberOfNodes = 2133 will also go under second if statement. I think we should use < instead of >

Reply:
I think the sign is correct but we need to flip the sequence of the checks (i.e., descending instead of ascending)

Reply:
Yes, can be any way round. However, we can possibly consolidate the multiple if statements returning same value.